kroxylicious-docs/docs/_modules/authorization/con-authorization-primitives.adoc

```diff
@@ -0,0 +1,837 @@
+:_mod-docs-content-type: CONCEPT
+
+
+[id='con-authorization-primitives-{context}']
+= Authorization Primitives
+
+Kafka clients send Kafka messages to perform operations on resources in Kafka.
+It is required to know the operations and resources to effectively configure authorization checks.
+This section lists these operations and resources, and then list the mapping between the Kafka messages and the operations and resources.
+
+== Resource type and operations
+
+This table lists the resources types and operations enforced by the authorization filter.
+
+[id='authorization-resource-types-and-operations-{context}']
+.Table of authorization primitives
+|===
+|Resource type|Operations
+
+|Topic
+|READ, WRITE, CREATE, DELETE ALTER, DESCRIBE, DESCRIBE_CONFIGS, ALTER_CONFIGS
+
+|Cluster
+|CREATE, ALTER, DESCRIBE, CLUSTER_ACTION, DESCRIBE_CONFIGS, ALTER_CONFIGS
+
+|===
+
+== Message, operation, and resource mapping
+
+
+The table below lists the kafka messages for which the authorization filter will make authorization enforcements.
+For each message, it shows the operation and resource type that will be tested.
+
+[id='authorization-message-mapping-{context}']
+.Table mapping the message to operation and resource
```

Comment:
I'm really sceptical that we want this level of detail in the documentation.

* Most users have approximately zero knowledge of, or interest in, the protocol.
* We've set the bar as "Kafka equivalence". Describing it like that that is much more meaningful to an end user, and much less of a maintenance burden when if comes to keeping the docs up to date with the code. 
* The actual logic which a broker applies is not actually quite as simple as these kinds of tables sometimes suggest.

Reply:
Agreed, I was having second thoughts myself.  I've removed the table.